src/core/engine/processors/kitchen_simulator.py

```diff
@@ -37,30 +37,39 @@ def assign_orders_to_available_chefs(self):
 
         if available_chef_ids:
 
-            order_in_turn_id = self.order_manager.get_queue_from_status(
-                OrderStatus.NEW_ORDER
-            )
-            order_to_be_assigned = list(
-                filter((lambda order: order.id == order_in_turn_id), orders_to_process)
-            )
-
-            if orders_validation_map[order_to_be_assigned[0].id]:
-                self._assign_to_chef_and_send_to_in_process(
-                    order_to_be_assigned[0], available_chef_ids[0]
+            if self.order_manager.get_queue_size(OrderStatus.NEW_ORDER) > 0:
```

Comment:
This check must not be added here, should be added in the order manager

Reply:
done.